units/commit_queue.go

```diff
@@ -288,22 +292,28 @@ func getModules(ctx context.Context, githubToken string, nextItem *commitqueue.C
 	return modulePRs, modulePatches, false, nil
 }
 
-func getPatchInfo(ctx context.Context, githubToken string, patchDoc *patch.Patch) (string, []patch.Summary, *model.Project, error) {
+func getPatchInfo(ctx context.Context, githubToken string, patchDoc *patch.Patch) (string, []patch.Summary, *model.Project, bool, error) {
 	patchContent, summaries, err := thirdparty.GetGithubPullRequestDiff(ctx, githubToken, patchDoc.GithubPatchData)
 	if err != nil {
-		return "", nil, nil, errors.Wrap(err, "can't get diff")
+		return "", nil, nil, false, errors.Wrap(err, "can't get diff")
 	}
 
 	// fetch the latest config file
-	config, err := validator.GetPatchedProject(ctx, patchDoc, githubToken)
+	yamlBytes, err := validator.GetPatchedProject(ctx, patchDoc, githubToken)
+	if err != nil {
+		return "", nil, nil, false, errors.Wrap(err, "can't get remote config file")
+	}
+
+	patchDoc.PatchedConfig = string(yamlBytes)
+	config, invalid, err := validator.ValidateProjectPatch(yamlBytes, patchDoc.Project)
 	if err != nil {
-		return "", nil, nil, errors.Wrap(err, "can't get remote config file")
+		return "", nil, nil, invalid, errors.Wrap(err, "invalid config patch")
 	}
 
-	return patchContent, summaries, config, nil
+	return patchContent, summaries, config, false, nil
 }
 
-func writePatchInfo(patchDoc *patch.Patch, config *model.Project, patchSummaries []patch.Summary, patchContent, projectID string) error {
+func writePatchInfo(patchDoc *patch.Patch, config *model.Project, patchSummaries []patch.Summary, patchContent string) error {
```

Comment:
maybe we should make the args a struct... I worry about twiddling the order of projectID and patch content (my default would be to put projectID first...)

Reply:
only three args now